src/cpp/session/modules/rmarkdown/NotebookAlternateEngines.cpp

```diff
@@ -338,6 +338,49 @@ Error executeStanEngineChunk(const std::string& docId,
    return Success();
 }
 
+Error executeSqlEngineChunk(const std::string& docId,
```

Comment:
I'm guessing that you need to tweak this code:

https://github.com/rstudio/rstudio/blob/e29a067e39385436a7c50873acf8752b5bff0dee/src/cpp/session/modules/rmarkdown/NotebookQueue.cpp#L232-L237

An R code chunk is finished executing when the console prompt appears following the execution of its last statement, whereas an alternate engine chunk is finished executing when it fires a `chunkExecCompleted` event. Your SQL engine both has an execution context (previously only extant for R chunks) **and** relies on the event to signal completion, which is excluding it from both means of advancement. You can probably just make the test above a little smarter (e.g. test the execution context itself).